src/main/java/seedu/address/model/InfoHandler.java

```diff
@@ -231,24 +250,38 @@ public VisitBook generateVisitsByPerson(Id personId) {
     }
 
     /**
-     * Generates a list of visits that are associated with the given list of location Ids.
-     * @param locationIds List of location Ids.
-     * @return List of visits that are associated with the location Ids.
+     * Generates a list of visits that are associated with the given visits.
+     * @param visitBook Visits provided.
+     * @return List of visits that are associated with the given visits.
      */
-    public VisitBook generateVisitsByLocationIds(List<Id> locationIds) {
+    public VisitBook generateAffectedVisits(VisitBook visitBook) {
         ReadOnlyVisitBook tempVisitBook = model.getVisitBook();
         VisitBook associatedVisits = new VisitBook();
-        for (Id locationId : locationIds) {
+        for (Visit givenVisit : visitBook.getVisitList()) {
             for (int i = 0; i < tempVisitBook.getVisitList().size(); i++) {
                 Visit visit = tempVisitBook.getVisitList().get(i);
-                if (visit.getLocation().getId().equals(locationId)) {
+                if (isAddableVisit(visit, givenVisit, associatedVisits)) {
                     associatedVisits.addVisit(visit);
                 }
             }
         }
         return associatedVisits;
     }
 
+    /**
+     * Checks if a visit is considered affected by a given visit and can be added to the list of affected visits.
+     * @param visit Visit to be checked.
+     * @param givenVisit Visit the above parameter is being checked against.
+     * @param associatedVisits List of visits already affected by the given visit.
+     * @return A boolean of whether the visit can be added or not.
+     */
```

Comment:
I personally feel that the explanation here could be a bit clearer as I do not understand what it means for a visit to be affected by a given visit.

Reply:
Added the criteria for being associated (originally affected). Hope it makes things clearer!